src/device.rs

```diff
@@ -110,6 +110,32 @@ impl<'a> Device<'a> {
         }
     }
 
+    /// Returns the first parent with a given subsystem and devtype
+    pub fn parent_with_subsystem_devtype<T: AsRef<OsStr>, U: AsRef<OsStr>>(&self, subsystem: T, devtype: Option<U>) -> ::Result<Option<Device>> {
```

Comment:
What do you think about splitting this into two functions?

```rust
pub fn parent_with_subsystem<...>(&self, subsystem: T) -> ::Result<Option<Device>> { /* ... */ }
pub fn parent_with_subsystem_devtype<...>(&self, subsystem: T, devtype: U) -> ::Result<Option<Device>> { /* ... */ }
```

Removing the `Option` from the argument removes ambiguity in type inference when `None` is known at compile-time. It would have the side-benefits of a more direct implementation and removing a runtime check on `devtype`.